src/ann_solo/reader.py

```diff
@@ -231,6 +295,233 @@ def get_version(self) -> str:
+            identifier = re.search('[0-9]+.', LibID.group(0)).group(0)
+        # Check if decoy
+        decoy = True if re.search('decoy', spectrum_metadata,
+                                  re.IGNORECASE) else False
+        ##Retrieve peptide & charge
+        peptide_Charge = spectrum_metadata.split('\n', 1)[0].split('/')
+        peptide = peptide_Charge[0].strip()
+        charge = int(peptide_Charge[1].strip())
+        ##Retrieve precurssor mass
+        perMZ = re.search('PrecursorMZ:\s?[0-9]+.[0-9]+', spectrum_metadata,
+                          re.IGNORECASE)
+        if perMZ:
+            perMZ = re.search('[0-9]+.[0-9]+', perMZ.group(0))
+        else:
+            perMZ = re.search('Parent=\s?[0-9]+.[0-9]+', spectrum_metadata,
+                              re.IGNORECASE)
+            perMZ = re.search('[0-9]+.[0-9]+', perMZ.group(0))
+        ##Retrieve MZ & Intensities
+        file = io.StringIO(spectrum)
+        MZ_Intensity = pd.read_csv(file, sep="\t", header=None)
+
+        spectrum = MsmsSpectrum(identifier, perMZ.group(0), charge,
+                                MZ_Intensity[0].to_numpy(copy=True),
+                                MZ_Intensity[1].to_numpy(copy=True),
+                                is_decoy=decoy)
+        spectrum.peptide = peptide
+
+        return spectrum
+
+    # For parallel processing
+    def _parse_batch_sptxt_spectra(self, raw_spectra: List[str], ids: List[
+        int]) -> Iterator[MsmsSpectrum]:
+        """
+        Takes a list of  raw spectra data retrieved from an sptxt file and
+        parses it to a structured object of type MsmsSpectrum.
+
+        Parameters
+        ----------
+        raw_spectrum : List[string]
+            The spectrum in a raw format.
+        identifier : List[int]
+            Incremented identifier of the spectrum in the library.
+
+        Returns
+        -------
+        Iterator[MsmsSpectrum]
+            An iterator of spectra in the given library file.
+        """
+        for spectrum in list(Parallel(n_jobs=cpu_count())(
+                delayed(_parse_sptxt_spectrum)(raw_spectrum, id) for
+                raw_spectrum, id in zip(raw_spectra, ids))):
+            yield spectrum
+
+    def parse_sptxt(self) -> Iterator[MsmsSpectrum]:
+        """
+        Open the sptxt spectra library file and parses it
+        to read all spectra.
+
+        Returns
+        -------
+        Iterator[MsmsSpectrum]
+            An iterator of spectra in the given library file.
+
+        """
+        with open(self._filename, 'rb') as file:
+            # Create a reference variable for Class RegexpTokenizer
+            tk = RegexpTokenizer('Name:\s?', gaps=True, flags=re.IGNORECASE)
+            mmapped_file = mmap.mmap(file.fileno(), 0, access=mmap.ACCESS_READ)
+            raw_spectra_list = []
+            for i, raw_spec in enumerate(
+                    tk.tokenize(mmapped_file.read().decode('utf-8')), 1):
+                if i % cpu_count():
```

Comment:
**question:** I don't understand this line. Why should some spectra be immediately parsed and others not? Can this section of code be simplified?